torchdynamo/eval_frame.py

```diff
@@ -360,6 +361,24 @@ def patch():
         if proxy_tensor is not None:
             proxy_tensor.dispatch_trace = disable(proxy_tensor.dispatch_trace)
 
+        optimizers = [
+            opt
+            for opt in torch.optim.__dict__.values()
+            if inspect.isclass(opt) and issubclass(opt, torch.optim.Optimizer)
+        ]
+
+        # disable profile hook
+        for opt in optimizers:
+            opt._cuda_graph_capture_health_check = disable(
+                opt._cuda_graph_capture_health_check
+            )
+            # disable future hooking
+            setattr(opt.step, "hooked", True)
+            # disable any currently set hooks
+            unwrapped_step = getattr(opt.step, "__wrapped__", None)
+            if unwrapped_step:
+                opt.step = unwrapped_step
```

Comment:
Do we nee to clean this up when we exit dynamo?

Reply:
Agreed, I can change this to on enter/exit

Reply:
I chatted with Jason about this a little more, and he said it's okay for now as longer term we will upstream these patches. Upstreaming this one will be trickier, but I think I can do it with a decorator. The reason I asked was because patching and unpatching on every enter and exit will have a tiny perf hit. In the end, the only person who will encounter this issue is someone who imports dynamo and then tries to interleave using dynamo with using `torch.autograd.profiler.record_function`.

Reply:
Yeah, its not a crazy important case - more about leaving things as we found them. I am find with us kicking this can down the road if you and Jason are.